File: src/main/java/net/kigawa/util/plugin/game/onigo/runnable/GameLimiter.java
package net.kigawa.util.plugin.game.onigo.runnable;

import net.kigawa.onigoplugin.OnigoPlugin;
import net.kigawa.util.plugin.game.onigo.Game;
import net.kigawa.util.plugin.game.stage.StageData;
import net.kigawa.util.plugin.game.stage.runnable.Limiter;

public class GameLimiter extends Limiter {
    Game game;


    public GameLimiter(OnigoPlugin OnigoPlugin, StageData stageData, Game game) {
        super(OnigoPlugin, game.getJoinPlayer(), stageData);
        this.game = game;
    }

    @Override
    public void onRun() {
        game.runnable();
    }
}

File: src/main/java/net/kigawa/onigoplugin/util/plugin/game/onigo/runnable/GameLimiter.kt
package net.kigawa.onigoplugin.util.plugin.game.onigo.runnable

import net.kigawa.onigoplugin.OnigoPlugin
import net.kigawa.onigoplugin.game.OnigoGame
import net.kigawa.onigoplugin.player.OnigoPlayerManager
import net.kigawa.onigoplugin.util.plugin.game.stage.StageData
import net.kigawa.onigoplugin.util.plugin.game.stage.runnable.Limiter

class GameLimiter(
  onigoPlugin: OnigoPlugin?, stageData: StageData?,
  private val game: OnigoGame,
  playerManager: OnigoPlayerManager
) : Limiter(
  onigoPlugin, playerManager.getAll(
    game, null, null
  ), stageData
) {
  override fun onRun() {
    game.runnable()
  }
}
